Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -148,6 +149,21 @@ final class ConfigGenerator: ConfigGenerating {
         pbxproj.add(object: variantBuildConfiguration)
         configurationList.buildConfigurations.append(variantBuildConfiguration)
     }
+    
+    private func update(buildSettings settings: inout [String: SettingValue], byApplying manifest: Target) {
+        if let deploymentTarget = manifest.deploymentTarget {
+            switch deploymentTarget {
+            case let .iOS(_, devices):
+                if manifest.platform == .iOS {
+                    settings["TARGETED_DEVICE_FAMILY"] = .string(devices.map { "\($0.rawValue)" }.joined(separator: ","))
```

Comment:
Is that a right place to do a validation like that? ☝️

Reply:
Ah yes, there could be some funny cases as there! 

Did you mean having `.macOS` `platform` with an `iOS` `deploymentTarget`?

```swift
// Project.swift

import ProjectDescription

let project = Project(name: "App",
                      targets: [
                        Target(name: "App",
                               platform: .macOS, // <--
                               deploymentTarget: .iOS("13.1", [.iphone])
```

Few options I could think of:

- Add a linter check to flag this error to users ([see `SettingsLinter`](https://github.com/tuist/tuist/blob/master/Sources/TuistGenerator/Linter/SettingsLinter.swift))
  - Some background: there are linters that apply some validation checks ahead of project generations to flag issues in form of warnings or errors

- Another is perhaps offer a different initialiser for `Target` that doesn't expose `platform`? We should be able to infer it given a `deploymentTarget` 

```swift
let project = Project(name: "App",
                      targets: [
                        Target(name: "App",
                               deploymentTarget: .iOS("13.1", [.iphone]), 
// ...
```

Reply:
> Did you mean having .macOS platform with an iOS deploymentTarget?

Yes, absolutely, sorry 😅 (I edited my comment).

> Add a linter check to flag this error to users (see SettingsLinter)

Hmm, so we could basically extend the target validation method like:

```
func lint(target: Target) -> [LintingIssue] { 
  ...
  if target.deploymentTarget.isDifferentThan(target.platform) { // we will come up with better method and naming 
     issues.append(LintingIssue(reason: "Inconsistency between deployment target \(target.deploymentTarget)" and platform \(target.platform) found, severity: .error))
  }
```

☝️ Obviously we will need to figure out more descriptive error with maybe some guides how to fix it. 

> Another is perhaps offer a different initialiser for Target that doesn't expose platform?
 
On the one hand in that ☝️way we could prevent users from putting an invalid project description at the API level 👍 On the other hand will this solution scale up well? I mean what if we supported more and more params, which would conflict each other? 

Let's think about that a bit more, because providing an extra param in public API brings compatibility maintenance struggle in the future 😉 

Thoughts?

Reply:
I'm more inclined towards linting that the platform and the deployment targets are compatible. Is it possible to have targets that without a deployment target? If not, I like the idea of merging those two arguments together but I'd do it in a major release. What do you think about adding an annotation to revisit this when we work on a major release? Something like:

```
// TODO_MAJOR_CHANGE: Merge deploymentTarget and platform arguments together.
```

Reply:
Moved a deployment target and a platform compatibility validation to `SettingsLinter` 👍